src/api/baggage.rs

```diff
@@ -1,22 +1,30 @@
 use crate::api::{Context, Key, KeyValue, Value};
+#[cfg(feature = "serialize")]
+use serde::{Deserialize, Serialize};
 use std::collections::{hash_map, HashMap};
 use std::iter::FromIterator;
 
 lazy_static::lazy_static! {
     static ref DEFAULT_BAGGAGE: Baggage = Baggage::default();
 }
 
+const MAX_KEY_VALUE_PAIRS: usize = 180;
+const MAX_BYTES_FOR_ONE_PAIR: usize = 4096;
+const MAX_LEN_OF_ALL_PAIRS: usize = 8192;
+
 /// A set of name/value pairs describing user-defined properties across systems.
 #[derive(Debug, Default)]
 pub struct Baggage {
-    inner: HashMap<Key, Value>,
+    inner: HashMap<Key, (Value, BaggageMetadata)>,
+    len: usize,
```

Comment:
May want to clarify that this is not the baggage len, maybe something like `kv_content_len` or some other name to disambiguate